GitCommands/Git/GitModule.cs

```diff
@@ -1007,14 +1006,15 @@ public GitRevision GetRevision(ObjectId? objectId = null, bool shortFormat = fal
 
                 // commit message is not re-encoded by git when format is given
                 // See also RevisionReader for parsing commit body
-                string body = ReEncodeCommitMessage(message);
-                revision.Body = body;
+                string body = ReEncodeCommitMessage(message).Replace('\v', '\n');
 
                 ReadOnlySpan<char> span = (body ?? "").AsSpan();
                 int endSubjectIndex = span.IndexOf('\n');
-                revision.Subject = endSubjectIndex >= 0
+                revision.HasMultiLineMessage = endSubjectIndex >= 0;
+                revision.Subject = revision.HasMultiLineMessage
                     ? span[..endSubjectIndex].TrimEnd().ToString()
                     : body ?? "";
+                revision.Body = revision.HasMultiLineMessage ? body : null;
```

Comment:
Do all users check `revision.HasMultiLineMessage`?
The following would be safer and reuses the string instance:
```suggestion
                revision.Body = body;
```

Reply:
HasMultiLineMessage is checked if the body is to be fetched (the reder only stores messages newer than 6 months) when a commit is selected.
If changed here, it should be changed everywhere. The handling here should probably use the string pool too for instance.
There are 19 accesses of Body, but I still prefer to not change this.

Reply:
My point is that the old code always sets body unconditionally:
```
                string? body = ReEncodeCommitMessage(message);
                revision.Body = body;
```
So `Body` could only be `null` if there was no `Subject`, too.

Setting `HasMultiLineMessage` was missing indeed.

Reply:
> My point is that the old code always sets Body unconditionally:

Correct, I reverted that

> There should not be significantly many identical commit messages. So I think pooling would rather have a negative performance impact.

That would be for author etc